examples/conversation_addons/tone_detection.js

```diff
@@ -0,0 +1,281 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+var watson = require('watson-developer-cloud');
+var Promise = require('bluebird');
+
+/**
+ * Thresholds for identifying meaningful tones returned by the Watson Tone Analyzer.  Current values are
+ * based on the recommendations made by the Watson Tone Analyzer at
+ * https://www.ibm.com/watson/developercloud/doc/tone-analyzer/understanding-tone.shtml
+ * These thresholds can be adjusted to client/domain requirements.
+ */
+var PRIMARY_EMOTION_SCORE_THRESHOLD = parseFloat("0.5");
+var LANGUAGE_HIGH_SCORE_THRESHOLD = parseFloat("0.75");
+var LANGUAGE_NO_SCORE_THRESHOLD = parseFloat("0.0");
+var SOCIAL_HIGH_SCORE_THRESHOLD = parseFloat("0.75");
+var SOCIAL_LOW_SCORE_THRESHOLD = parseFloat("0.25");
+
+/**
+ * Labels for the tone categories returned by the Watson Tone Analyzer
+ */
+var EMOTION_TONE_LABEL = "emotion_tone";
+var LANGUAGE_TONE_LABEL = "language_tone";
+var SOCIAL_TONE_LABEL = "social_tone";
+
+/**
+ * Public functions for this module
+ */
+module.exports = {
+    updateUserTone,
+    invokeToneAsync
+};
+
+/**
+ * invokeToneAsync is an asynchronous function that calls the Tone Analyzer service and returns a Promise 
+ * @param conversationPayload json object returned by the Watson Conversation Service
+ * @param tone_analyzer an instance of the Watson Tone Analyzer service
+ * @returns a Promise for the result of calling the tone_analyzer with the conversationPayload
+ * (which contains the user's input text) 
+ */
+function invokeToneAsync(conversationPayload, tone_analyzer) {
+  return new Promise(
+      function (resolve, reject){
+        tone_analyzer.tone(
+            {text: conversationPayload.input.text},
+            (error, data) => {
+              if (error){
+                reject(error);
+              }else {
+                resolve(data);
+              }
+            });
+      });
+};
+
+/**
+ * updateUserTone processes the Tone Analyzer payload to pull out the emotion, language and social
+ * tones, and identify the meaningful tones (i.e., those tones that meet the specified thresholds).  
+ * The conversationPayload json object is updated to include these tones.
+ * @param conversationPayload json object returned by the Watson Conversation Service
+ * @param toneAnalyzerPayload json object returned by the Watson Tone Analyzer Service
+ * @returns conversationPayload where the user object has been updated with tone information from the toneAnalyzerPayload 
+ */
+function updateUserTone (conversationPayload, toneAnalyzerPayload) {
+  var emotionTone = null;
+  var languageTone = null;
+  var socialTone = null;
+
+  if(typeof conversationPayload.context === 'undefined')
```

Comment:
Open brackets in the same line the `if` is declared. See our [CONTRIBUTING](https://github.com/watson-developer-cloud/node-sdk/blob/master/.github/CONTRIBUTING.md) guidelines